### PR TITLE
add dry run scripts for notices report and renewal application resubmission

### PR DESCRIPTION
**Pull request by TristanB17**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/188213034

# A brief description of the changes

Current behavior: No way to determine which Person documents have received notices. Resubmit applications script not present.

New behavior: Ports a script for generating a report of all Person documents which have received notices between a set of dates. Resubmit application script originally used in-console added to source control.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: app/domain/operations/reports/generate_notices_report.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+      def fetch_notice_report_file_name
+        "notices_list_#{TimeKeeper.date_of_record.strftime('%Y_%m_%d')}.csv"
+      end
+
+      def fetch_notice_report_headers
+        ['HBX ID', 'Notice Title', 'Notice Code', 'Date']
+      end
+
+      def notices_title_code_mapping
```
Comment (vkghub) [inline app/domain/operations/reports/generate_notices_report.rb:52]: - Fetching every notice from DB takes lot of time for IVL dry run scenarios.
- Instead we can add constant tags to fetch specific data for every stage of dry run, may be something like this.

```suggestion
      OE_ELIGIBILITY_NOTICES_TITLE_MAPPING = {}.freeze
      OE_ENROLLMENT_NOTICES_TITLE_MAPPING = {}.freeze
      NON_OE_ELIGIBILITY_NOTICES_TITLE_MAPPING = {}.freeze
      DR_NOTICES_TITLE_MAPPING = {}.freeze
```

File: components/financial_assistance/app/domain/financial_assistance/operations/applications/aptc_csr_credit_eligibilities/renewals/resubmit.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+              Success(params)
+            end
+
+            # Fetches the current year applications that are eligible for renewal in the given renewal year.
+            # The current year is assumed to be the year before the given renewal year.
+            #
+            # @param params [Hash] The parameters to filter applications.
+            # @option params [Integer] :renewal_year The renewal year to filter applications.
+            # @return [Success, Failure] Returns a Success object with eligible applications if found, otherwise returns a Failure object with an error message.
+            def applications_for_families(params, family_ids = ::HbxEnrollment.individual_market.enrolled.current_year.distinct(:family_id))
```
Comment (vkghub) [inline components/financial_assistance/app/domain/financial_assistance/operations/applications/aptc_csr_credit_eligibilities/renewals/resubmit.rb:45]: Only this command we can use to fetch application to resubmit them. we can ignore all the other queries and if conditions in this operation.
```
FinancialAssistance::Application.no_timeout.by_year(@renewal_year).where(:family_id.in => renewal_eligible_family_ids, :aasm_state.in => ["submitted", "renewal_draft"])
```

File: app/domain/operations/reports/generate_notices_report.rb (hunk 1 above)
Comment (TristanB17) [reply app/domain/operations/reports/generate_notices_report.rb:52]: Made [separate ticket](https://www.pivotaltracker.com/story/show/188260680) to refactor this operation

Comment (mdkaraman) [inline app/domain/operations/reports/generate_notices_report.rb:52]: Looks like we need to update these mappings.  The titles and codes are different between clients.  These were the mappings used last year for the ME client:
```
title_code_mapping = {"Welcome to CoverME.gov!"=>"IVLMWE", "Your Plan Enrollment"=>"IVLENR", "Your Eligibility Results - Tax Credit"=>"IVLERA", "Your Eligibility Results - MaineCare or Cub Care"=>"IVLERM", "Your Eligibility Results - Marketplace Health Insurance"=>"IVLERQ", "Your Eligibility Results - Marketplace Insurance"=>"IVLERU", "Open Enrollment - Tax Credit"=>"IVLOEA", "Open Enrollment - Update Your Application"=>"IVLOEM", "Your Eligibility Results - Health Coverage Eligibility"=>"IVLOEQ", "Open Enrollment - Marketplace Insurance"=>"IVLOEU", "Your Eligibility Results Consent or Missing Information Needed"=>"IVLOEG", "Find Out If You Qualify For Health Insurance On CoverME.gov"=>"IVLMAT", "Your Plan Enrollment for 2022"=>"IVLFRE", "Action Needed - Submit Documents"=>"IVLDR0", "Reminder - You Must Submit Documents"=>"IVLDR1", "Don't Forget - You Must Submit Documents"=>"IVLDR2", "Don't Miss the Deadline - You Must Submit Documents"=>"IVLDR3", "Final Notice - You Must Submit Documents"=>"IVLDR4"}
```

**Review by mdkaraman — CHANGES_REQUESTED** [review]

File: app/domain/operations/reports/generate_notices_report.rb (hunk 1 above)
Comment (mdkaraman) [reply app/domain/operations/reports/generate_notices_report.rb:52]: @TristanB17 thanks for making the update!  I'll go ahead and resolve this conversation.

File: app/assets/javascripts/plan_shopping.js.erb [hunk 3]
```diff
@@ -377,6 +377,7 @@ $(function() {
         },
         success: function(data){
           response = JSON.parse(JSON.stringify(data));
+          console.log(response.checkbook_url);
```
Comment (mdkaraman) [inline app/assets/javascripts/plan_shopping.js.erb:380]: Looks like this change may have snuck in unintentionally?

**Review by mdkaraman — DISMISSED** [review]
Looks good!  Adds scripts to the code base that we use in the annual dry run processes.

**Review by ymhari — DISMISSED** [review]

**Review by ymhari — APPROVED** [review]

**Review by mdkaraman — APPROVED** [review]

**Pull request merged**